historical_system_profiles/archiver.py

```diff
@@ -92,19 +100,24 @@ def _archive_profile(data, ptc, logger, notification_service):
     # profiles in the app via captured_date.
 
     if captured_date and db_interface.is_profile_recorded(
-        captured_date, host["id"], account, org_id
+        captured_date, inventory_id, account, org_id
     ):
         logger.info(
             "profile with date %s is already recorded for %s, using account id: %s, org_id: %s"
-            % (captured_date, host["id"], account, org_id)
+            % (captured_date, inventory_id, account, org_id)
         )
         _record_duplicate_message(host, request_id, ptc)
     else:
+        # remove keys we do not care about
+        if groups:
```

Comment:
This `if ` can be handled within the function.

Reply:
:-)